_docs/setup/kubernetes/mesh-expansion.md

```diff
@@ -31,61 +31,76 @@ separate documents will cover these advanced configurations.
 
 An example script to help configure a machine is available in [install/tools/setupIstioVM.sh](https://raw.githubusercontent.com/istio/istio/master/install/tools/setupIstioVM.sh).
 You should customize it based on your provisioning tools and DNS requirements.
 
 ### Preparing the Kubernetes cluster for expansion
 
-* Setup internal load balancers for Kube DNS, Pilot, Mixer and CA. This step is specific to
-each cluster, you may need to add annotations.
+* Setup Internal Load Balancers (ILBs) for Kube DNS, Pilot, Mixer and CA. This step is specific to
+each cloud provider, you may need to edit annotations.
+
+> The yaml part of the 0.2.7 distribution has a typo with the wrong
+namespace for the DNS ILB, use
+[this one](https://raw.githubusercontent.com/istio/istio/master/install/kubernetes/mesh-expansion.yaml),
+likewise for the setupMeshEx.sh, please use the latest above or from cloning [GitHub.com/istio/istio](https://github.com/istio/istio/)
 
-```bash
-install/tools/setupMeshEx initCluster
-```
-or
 ```
-kubectl apply -f install/kubernetes/meshex.yaml
+kubectl apply -f install/kubernetes/mesh-expansion.yaml
 ```
 
 * Generate the Istio 'cluster.env' configuration to be deployed in the VMs. This file contains
 the cluster IP address ranges to intercept.
 
 ```bash
-install/tools/setupMeshEx generateConfigs MY_CLUSTER_NAME
+export GCP_OPTS="--zone MY_ZONE --project MY_PROJECT"
+install/tools/setupMeshEx.sh generateClusterEnv MY_CLUSTER_NAME
 ```
 
-Example generated files:
+Example generated file:
+
+```bash
+cat cluster.env
+```
+```
+ISTIO_SERVICE_CIDR=10.63.240.0/20
+```
 
-   ```bash
-   cat /usr/local/istio/proxy/cluster.env
-   ```
-   ```
-   ISTIO_SERVICE_CIDR=10.23.240.0/20
-   ```
 * Generate DNS configuration file to be used in the VMs. This will allow apps on the VM to resolve
 cluster service names, which will be intercepted by the sidecar and forwarded.
 
 ```bash
-install/tools/setupMeshEx generateConfigs MY_CLUSTER_NAME
+# Make sure your kubectl context is set to your cluster
+install/tools/setupMeshEx.sh generateDnsmasq
 ```
 
-Example generated files:
+Example generated file:
 
-   ```bash
+```bash
 cat /etc/dnsmasq.d/kubedns
 ```
 ```
-   server=/svc.cluster.local/10.128.0.6
-   address=/istio-mixer/10.128.0.7
-   address=/mixer-server/10.128.0.7
-   address=/istio-pilot/10.128.0.5
-   address=/istio-ca/10.128.0.8
-  ```
+server=/svc.cluster.local/10.150.0.7
+address=/istio-mixer/10.150.0.8
+address=/istio-pilot/10.150.0.6
+address=/istio-ca/10.150.0.9
+address=/istio-mixer.istio-system/10.150.0.8
+address=/istio-pilot.istio-system/10.150.0.6
+address=/istio-ca.istio-system/10.150.0.9
+```
 
 ### Setting up the machines
 
+There is a script that copies and install the setup you can use as an example:
+```bash
+# Check what the script does to see that it meets your need:
+# On a mac either brew install base64 or set BASE64_DECODE="/usr/bin/base64 -D"
```

Comment:
Check what the script does to see if it meets your needs.
On a Mac, either brew...